src/main/java/org/junitpioneer/jupiter/EnvironmentVariableUtils.java

```diff
@@ -16,64 +16,66 @@
 import java.util.Map;
 import java.util.function.Consumer;
 
+import org.junit.jupiter.api.extension.ExtensionConfigurationException;
+
 /**
  * This class modifies the internals of the environment variables map with reflection.
  * Warning: If your {@link SecurityManager} does not allow modifications, it fails.
  */
 public class EnvironmentVariableUtils {
 
 	/**
-	 * Set the values of an environment variables.
+	 * Set a value of an environment variable.
 	 *
-	 * @param entries with name and new value of the environment variables
+	 * @param name  of the environment variable
+	 * @param value of the environment variable
 	 */
-	public static void set(Map<String, String> entries) {
-		modifyEnvironmentVariables(map -> map.putAll(entries));
+	public static void set(String name, String value) {
+		modifyEnvironmentVariables(map -> map.put(name, value));
 	}
 
 	/**
-	 * Clears environment variables.
+	 * Set the values of an environment variables.
 	 *
-	 * @param names of the environment variables.
+	 * @param entries with name and new value of the environment variables
 	 */
-	public static void clear(Collection<String> names) {
-		modifyEnvironmentVariables(map -> names.forEach(map::remove));
+	public static void set(Map<String, String> entries) {
+		modifyEnvironmentVariables(map -> map.putAll(entries));
 	}
 
 	/**
-	 * Set a value of an environment variable.
+	 * Clear an environment variable.
 	 *
-	 * @param name  of the environment variable
-	 * @param value of the environment variable
+	 * @param name of the environment variable
 	 */
-	public static void set(String name, String value) {
-		modifyEnvironmentVariables(map -> map.put(name, value));
+	public static void clear(String name) {
+		modifyEnvironmentVariables(map -> map.remove(name));
 	}
 
 	/**
-	 * Clear an environment variable.
+	 * Clears environment variables.
 	 *
-	 * @param name of the environment variable
+	 * @param names of the environment variables.
 	 */
-	public static void clear(String name) {
-		modifyEnvironmentVariables(map -> map.remove(name));
+	public static void clear(Collection<String> names) {
+		modifyEnvironmentVariables(map -> names.forEach(map::remove));
 	}
 
 	private static void modifyEnvironmentVariables(Consumer<Map<String, String>> consumer) {
 		try {
 			tryProcessEnvironmentClassFallbackSystemEnvClass(consumer);
 		}
-		catch (NoSuchFieldException e) {
-			throw new RuntimeException("Could not modify environment variables");
+		catch (ReflectiveOperationException ex) {
+			throw new ExtensionConfigurationException("Could not modify environment variables", ex);
```

Comment:
Keep in mind that `ex` is always form the second method never from the first. The best would proably if we can save also the exception from the first method and if both fail somehow add both exceptions here but I do not know a way do that. Maybe we should change the message that reflects this.
'Could not modify environment variables. The attached cause exception is only from the second of the two methods.' 
what do you think?

Reply:
Instead of the cause we could also use to suppressed too better represent that they are on the same level:

```
    public static void main(String[] args) {
        try {
            try {
                throw new ReflectiveOperationException("First try");
            } catch (ReflectiveOperationException e) {
                try {
                    throw new ReflectiveOperationException("Second try");
                } catch (ReflectiveOperationException ex) {
                    RuntimeException runtimeException = new RuntimeException("all");
                    runtimeException.addSuppressed(e);
                    runtimeException.addSuppressed(ex);
                    throw runtimeException;
                }
            }
        } catch (RuntimeException runtimeE)
        {
            runtimeE.printStackTrace();
        }
    }
```
Output:
```
java.lang.RuntimeException: all
	at org.junitpioneer.Test.main(Test.java:13)
	Suppressed: java.lang.ReflectiveOperationException: First try
		at org.junitpioneer.Test.main(Test.java:8)
	Suppressed: java.lang.ReflectiveOperationException: Second try
		at org.junitpioneer.Test.main(Test.java:11)
```

Reply:
(A) Great catch about the first exception, which went missing. Adding it as suppressed to the second exception looks right to me because the code that lead to the second exception did indeed suppress the first.

(B) I would still add the second exception as cause to the final (third) one because that causal chain represents well what actually happened.

As far as I can tell the code still does (A) and I would keep it that way.